### ISSUE TITLE
DropdownButton and BottomNavigationBar not working with JAWs

### ISSUE THREAD

**Issue by ipainchaud** [flutter/flutter#117997]
# DropdownButton and BottomNavigationBar not working with JAWS

We noticed recently that JAWs is not able to activate the BottomNavigationBar or to select a DropdownButton Item anymore. We have tested with NVDA, VoiceOver (MacOs & iOS), TalkBack and JAWs and the issue can only be reproduced with JAWs (tested with JAWs 2021, 2022 and 2023).

### Code Samples
[dartpad gist to reproduce the BottomNavigationMenu issue](https://dartpad.dev/?id=8aae2e92bdf74e6b20767ab2244e9024)
[dartpad gist to reproduce the DropdownButton issue](https://dartpad.dev/?id=01d02c053384d7d97cb9348dfcc9dc34)

1. Use JAWs to navigate through the flutter generated UI. 
2. Navigate to an unselected Dropdown button Item OR Bottom Navigation Item you want to select.
3. Press enter (or space) to select the desired menu.

Actual result: The dropdown button item OR bottom navigation item is not selected. The `onTap` callback of the DropdownMenuItem doesn't even get called.

Expected result: JAWs should consider the dropdown bottom items OR bottom navigation items as a buttons and be able to select different values.

As both these cases can be reproduced through the provided link above in a dartpad environment, I didn't attach `flutter doctor`'s ouput.

**Comment by huycozy** [comment]
Hi @ipainchaud, thanks for filing the issue.
At your 3rd step, after pressing `enter` or `space`, is the item selected?  Can you provide a recorded video for this so that we can see the detailed steps to reproduce this?

I'm testing with my NVDA and JAWS (2023), and it seems JAWS is better than NVDA, both of them are also not selecting items after pressing these keys though.


https://user-images.githubusercontent.com/104349824/210751944-339f9d74-306d-4db6-b843-a100044a8fbc.mp4

**Comment by ipainchaud** [comment]
When navigating through an application with a screen reader, you should use tab or some arrow combination. In the following videos, I navigate using tab. The focus has to be in the right window otherwise it may be complicated to get to the button you want to test. In the attached videos, I click on the right pane and then tab. Then I use "enter" and "space" to select the wanted items. I repeated the same sequence in both videos.

One thing I didn't show in both videos is the use of these samples with the mouse and the Screen Reader enabled. This works in both cases.

https://user-images.githubusercontent.com/5298744/210842957-4c3495e1-4209-42c6-a2f6-8f0a058ecc06.mp4


https://user-images.githubusercontent.com/5298744/210842970-58d4c7ff-2b5c-4d7d-9873-63b858cb51cc.mp4

**Comment by vsomayaji** [comment]
The `BottomNavigationBar` issue may be due to the [`_BottomNavigationTile`](https://github.com/flutter/flutter/blob/36a1caf299237330a70e6493df339c95b6682204/packages/flutter/lib/src/material/bottom_navigation_bar.dart#L532-L560) using an `InkResponse` without `button` semantics, similar to https://github.com/flutter/flutter/issues/45823. 

The screen recording below demonstrates using JAWS to interact with the attached code sample. Pressing **Enter** on an `InkResponse` without button semantics does not trigger `onTap`. Interestingly, VoiceOver doesn't have this issue. 🤷🏾 

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();

  // Auto-enable accessibility for our Blind and Low Vision customers.
  RendererBinding.instance.setSemanticsEnabled(true);

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  static const String _title = 'InkResponse A11y Issue';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(
          title: const Text(_title),
        ),
        body: MyStatefulWidget(),
      ),
    );
  }
}

class MyStatefulWidget extends StatefulWidget {
  const MyStatefulWidget({super.key});

  @override
  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
}

class _MyStatefulWidgetState extends State<MyStatefulWidget> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(20),
      child: Center(
        child: Column(
          children: <Widget>[
            ElevatedButton(
              onPressed: _incrementCounter,
              child: const Text('ElevatedButton'),
            ),
            const SizedBox(height: 10),
            InkResponse(
              onTap: _incrementCounter,
              child: const Text('InkResponse'),
            ),
            const SizedBox(height: 10),
            Semantics(
              button: true,
              child: InkResponse(
                onTap: _incrementCounter,
                child: const Text('InkResponse with button semantics'),
              ),
            ),
            const SizedBox(height: 10),
            const Text('You have clicked a button this many times:'),
            Text('$_counter', style: Theme.of(context).textTheme.headline4),
          ],
        ),
      ),
    );
  }
}
```
</details>

[InkResponseA11yIssue.webm](https://user-images.githubusercontent.com/1007109/210871594-aea4b9e7-552d-476e-9572-36542cb34863.webm)

**Comment by huycozy** [comment]
@ipainchaud thanks for adding the detailed steps. I also can see the issue when using `tab` key to navigate and select item, the selected value is not updated to BottomNavigationBar or DropdownButton.

I also did some tests on Window app, the result seems to be better, there is a side issue of mouse hovering though.

<details>
<summary>Demo(Window app)</summary>


https://user-images.githubusercontent.com/104349824/210939356-03356f08-e13e-4c8a-b456-cd4bc1eb341c.mp4


https://user-images.githubusercontent.com/104349824/210939370-a9887d6a-5fd0-4e9a-8b3f-a02c9a4705fe.mp4


</details>

Labeling the issue for further investigation. Summary of the issue as below:

#### On Web
- JAWS
    - Selected value is not updated when selecting an item by `enter` or `space` key ❌
    - It does not read when hovering the item with mouse ❌
- NVDA
    - Selected value is updated when selecting an item by `enter` or `space` key ✅
    - It does not read when hovering the item with mouse  ❌


#### On Window
- JAWS
    - Selected value is updated when selecting an item by `enter` or `space` key ✅
    - It does not read when hovering the item with mouse ❌
- NVDA
    - Selected value is updated when selecting an item by `enter` or `space` key ✅
    - It does read when hovering the item with mouse  ✅

✅: No Issue ❌: Issue reproduced

<details>
<summary>flutter doctor -v (stable & master)</summary>

```bash
[√] Flutter (Channel stable, 3.3.10, on Microsoft Windows [Version 10.0.19045.2364], locale en-US)
    • Flutter version 3.3.10 on channel stable at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (4 days ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

Checking Android licenses is taking an unexpectedly long time...[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.2.32630.192
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.73.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.52.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2364]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 107.0.1418.35

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.7.0-14.0.pre.52, on Microsoft Windows [Version 10.0.19045.2364], locale en-US)
    • Flutter version 3.7.0-14.0.pre.52 on channel master at C:\WIP\flutter_master
    ! Warning: `flutter` on your path resolves to C:\WIP\flutter\bin\flutter, which is not inside your current
      Flutter SDK checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your
      path.
    ! Warning: `dart` on your path resolves to C:\WIP\flutter\bin\dart, which is not inside your current Flutter SDK
      checkout at C:\WIP\flutter_master. Consider adding C:\WIP\flutter_master\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 507062032f (5 hours ago), 2023-01-06 01:13:18 +0100
    • Engine revision 5bd90d6e77
    • Dart version 3.0.0 (build 3.0.0-85.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git"
      directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)
    • Visual Studio at D:\DOWNLOADWORK\VS2022
    • Visual Studio Community 2022 version 17.2.32630.192
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2364]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.54

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```
</details>

**Comment by yjbanov** [comment]
@hangyujin Is there more to do to close this issue? PRs https://github.com/flutter/flutter/pull/150578 and https://github.com/flutter/flutter/pull/150576 seem to cover both the dropdown and the bottom nav bar.

**Comment by hangyujin** [comment]
yes i think this is fixed.
